src/pip/_internal/network/session.py

```diff
@@ -369,32 +385,38 @@ def __init__(
             # Add a small amount of back off between failed requests in
             # order to prevent hammering the service.
             backoff_factor=0.25,
-        )  # type: ignore
+        )
 
         # Our Insecure HTTPAdapter disables HTTPS validation. It does not
         # support caching so we'll use it for all http:// URLs.
         # If caching is disabled, we will also use it for
         # https:// hosts that we've marked as ignoring
         # TLS errors for (trusted-hosts).
-        insecure_adapter = InsecureHTTPAdapter(max_retries=retries)
+        insecure_adapter = InsecureHTTPAdapter(max_retries=retry)
 
         # We want to _only_ cache responses on securely fetched origins or when
         # the host is specified as trusted. We do this because
         # we can't validate the response of an insecurely/untrusted fetched
         # origin, and we don't want someone to be able to poison the cache and
         # require manual eviction from the cache to fix it.
         if cache:
-            secure_adapter = CacheControlAdapter(
-                cache=SafeFileCache(cache),
-                max_retries=retries,
-                ssl_context=ssl_context,
+            secure_adapter = cast(
+                HTTPAdapter,
+                CacheControlAdapter(
+                    cache=SafeFileCache(cache),
+                    max_retries=retry,
+                    ssl_context=ssl_context,
+                ),
             )
-            self._trusted_host_adapter = InsecureCacheControlAdapter(
-                cache=SafeFileCache(cache),
-                max_retries=retries,
+            self._trusted_host_adapter = cast(
```

Comment:
Rather than these casts, I wonder if you can change network/session (https://github.com/pypa/pip/blob/a4b40f62332ccb3228b12cc5ae1493c75177247a/src/pip/_internal/network/session.py#L30C18-L30C30) to claim that `from pip._vendor.cachecontrol import CacheControlAdapter as _BaseCacheControlAdapter` is really `cachecontrol import CacheControlAdapter as _BaseCacheControlAdapter` when type-checking?